lib/repositories/switch-repository.ts

```diff
@@ -84,4 +84,20 @@ export class SwitchRepository implements BaseSwitchRepository {
     });
     console.log(JSON.stringify(res, null, 2));
   }
+
+  static getKey(trade: SynthSwitchTrade): string {
+    const { inputToken, inputTokenChainId, outputToken, outputTokenChainId, type } = trade;
+    return `${inputToken}#${inputTokenChainId}#${outputToken}#${outputTokenChainId}#${type}`;
+  }
+
+  static parseKey(key: string): SynthSwitchTrade {
+    const [inputToken, inputTokenChainId, outputToken, outputTokenChainId, type] = key.split('#');
```

Comment:
nit: maybe throw with a meaningful error msg here if the `key.split` doesn't fit the format of `[inputToken, inputTokenChainId, outputToken, outputTokenChainId, type]`. Could just check key length ?

Reply:
hm length is hard because chainIds can be different lengths

Reply:
I can check for definedness though at least